botocore/handlers.py

```diff
@@ -472,6 +472,13 @@ def check_openssl_supports_tls_version_1_2(**kwargs):
         pass
 
 
+def cloudsearchdomain_search_post(request, operation_name, **kwargs):
```

Comment:
The name is sort of ambiguous is weird here. The functionality is not cloudsearchdomain specific. It could be used in other places possibly. Be nice if we can be explicit in the function name of what you are doing. Something a long the lines of switching the get to post and putting the querysting into the body. Also I do not think you need ``operation_name`` in the signature.